### PR TITLE
Revert make currency id required in invoice #3491

### LINKED ISSUE #3491: Revert make currency_id required in invoice
## What went wrong? 😲
This change makes it so we can't upgrade from < 1.7.0 version since currency table doesn't exist until then.

## How to Reproduce 🔨

Steps to reproduce the behaviour:

## Your environment 🌱

- Version: 
- mSupply version: 

- Platform:

- Database type:

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #3491 

# 👩🏻‍💻 What does this PR do? 
Make currency_id on invoice optional again. Have left some of the tests as is (: 

# 🧪 How has/should this change been tested? 
- [ ] Make sure tests are still running fine

### CODE REVIEW

File: server/service/src/sync/translations/invoice.rs [hunk 1]
```diff
@@ -97,8 +97,7 @@ pub struct LegacyTransactRow {
     #[serde(rename = "currency_ID")]
     #[serde(deserialize_with = "empty_str_as_option_string")]
     pub currency_id: Option<String>,
-    pub currency_rate: Option<f64>,
-
+    pub currency_rate: f64,
```
Comment (clemens-msupply) [inline server/service/src/sync/translations/invoice.rs:100]: So this one is always set in mSupply?

File: client/packages/invoices/src/Prescriptions/api/operations.generated.ts [hunk 2]
```diff
@@ -4,104 +4,8 @@ import { GraphQLClient } from 'graphql-request';
...
...nt: number, nodes: Array<{ __typename: 'InvoiceLineNode', id: string, type: Types.InvoiceLineNodeType, batch?: string | null, expiryDate?: string | null, numberOfPacks: number, packSize: number, invoiceId: string, sellPricePerPack: number, note?: string | null, totalBeforeTax: number, totalAfterTax: number, taxPercentage?: number | null, item: { __typename: 'ItemNode', id: string, name: string, code: string, unitName?: string | null }, location?: { __typename: 'LocationNode', id: string, name: string, code: string, onHold: boolean } | null, stockLine?: { __typename: 'StockLineNode', id: string, itemId: string, batch?: string | null, availableNumberOfPacks: number, totalNumberOfPacks: number, onHold: boolean, sellPricePerPack: number, packSize: number, expiryDate?: string | null, item: { __typename: 'ItemNode', name: string, code: string } } | null }> }, patient?: { __typename: 'PatientNode', id: string, name: string, code: string, isDeceased: boolean } | null, clinician?: { __typename: 'ClinicianNode', id: string, firstName?: string | null, lastName: string } | null, currency?: { __typename: 'CurrencyNode', id: string, code: string, rate: number, isHomeCurrency: boolean } | null };
```
[the remark this answers is not in the document]
Comment (roxy-dao) [reply client/packages/invoices/src/Prescriptions/api/operations.generated.ts:8]: Maybe someone did manually format it 🤷🏻 I just pressed generate XD

File: server/service/src/sync/translations/invoice.rs (hunk 1 above)
Comment (roxy-dao) [reply server/service/src/sync/translations/invoice.rs:100]: It defaults to 1.0 anyways when created. 🤷🏻 which is the rate for home currencies

**Review by clemens-msupply — APPROVED** [review]
Looks good thanks

**Pull request merged**